clash-starters/clash-example-project/README.md

```diff
@@ -0,0 +1,137 @@
+
+# Using this project
+There's a number of ways to build this project on your machine. The recommended way of doing so is using _Stack_, whose instructions will follow directly after this section. Depending on your or your organization's needs, you might want to select another tool though. If you need help deciding, scroll down to [There's so many! Which should I pick?](#theres-so-many-which-should-i-pick).
+
+
+## Stack (Windows, Linux, MacOS) [recommended]
+Install Stack using your package manager or refer to the [How to install](https://docs.haskellstack.org/en/stable/README/#how-to-install) section of the [Stack manual](https://docs.haskellstack.org/en/stable/README/).
+
+Build the project with:
+
+```bash
+stack build
+```
+
+To run the tests defined in `tests/`, use:
+
+```bash
+stack test
+```
+
+To compile the project to VHDL, run:
+
+```bash
+stack run clash -- Example.Project --vhdl
+```
+
+
+## Snap (Linux)
+Linux users can use _snap_ to install Clash on their machines. See [snapcraft.io/clash](https://snapcraft.io/clash) how to do so. After installing Clash through snap, you can compile this project using:
+
+```bash
+clash.cabal update
+clash.cabal build
+```
+
+You only have to run the update command once. After that, you can keep rebuilding your project by running the build command.
+
+To run the tests defined in `tests/`, use:
+
+```bash
+clash.cabal run test-library
+```
+
+To compile the project to VHDL, run:
+
+```bash
+clash.cabal build --write-ghc-environment-files=always
+clash Example.Project --vhdl
+```
+
+Clash will look for a function called `topEntity` in the module you specify and compile that to HDL.
+
+<!-- omit in toc -->
+### Notes on using snap for your project
+
+1. By default, snaps will update to the latest stable release of Clash. This is a problem, because new (major) releases might break your design. To mitigate this, the next major Clash release (1.4) will introduce a [Snap channel](https://snapcraft.io/docs/channels) that will allow you to keep using 1.2.
+2. The version of `clash-{prelude,lib,ghc}` your project depends on must exactly match the one supplied in the snap. For this reason, we recommend specifying a range of Clash versions your project can work with. This way, cabal will automatically prefer the version installed int he snap.
+
+## Cabal (Linux, MacOS)
+**The following instructions only work for Cabal >=3.0, GHC >=8.4, and Clash >=1.2.2.**
+
+First, update your cabal package database:
+
+```bash
+cabal update
+```
+
+You only have to run the update command once. After that, you can keep rebuilding your project by running the build command:
+
+```bash
+cabal build
+```
+
+To run the tests defined in `tests/`, use:
+
+```bash
+cabal run test-library
+```
+
+To compile the project to VHDL, run:
+
+```bash
+cabal run clash -- Example.Project --vhdl
+```
+
+
+## Nix (Linux, MacOS)
+To be done.
```

Comment:
To be done before this PR or after?

Reply:
After I think. My guess is that it will take some time to do a write-up.